heartbeat/monitors/active/http/http.go

```diff
@@ -58,44 +58,44 @@ func create(
 		compression := config.Check.Request.Compression
 		enc, err = getContentEncoder(compression.Type, compression.Level)
 		if err != nil {
-			return nil, err
+			return nil, -1, err
 		}
 
 		buf := bytes.NewBuffer(nil)
 		err = enc.Encode(buf, bytes.NewBufferString(config.Check.Request.SendBody))
 		if err != nil {
-			return nil, err
+			return nil, -1, err
 		}
 
 		body = buf.Bytes()
 	}
 
 	validator := makeValidateResponse(&config.Check.Response)
 
-	jobs := make([]monitors.Job, len(config.URLs))
+	jobs = make([]monitors.Job, len(config.URLs))
 
 	if config.ProxyURL != "" {
 		transport, err := newRoundTripper(&config, tls)
 		if err != nil {
-			return nil, err
+			return nil, -1, err
 		}
 
 		for i, url := range config.URLs {
 			jobs[i], err = newHTTPMonitorHostJob(url, &config, transport, enc, body, validator)
 			if err != nil {
-				return nil, err
+				return nil, -1, err
 			}
 		}
 	} else {
 		for i, url := range config.URLs {
 			jobs[i], err = newHTTPMonitorIPsJob(&config, url, tls, enc, body, validator)
 			if err != nil {
-				return nil, err
+				return nil, -1, err
```

Comment:
Why did you pick -1 and not 0?

Reply:
I'm not partial to -1 or 0. I picked -1  because it's obviously a nonsense value in the error case. In any event, in all these situations the value should be discarded by any callers.

Reply:
My thinking: The method returns the number of endpoints it created. In the case of error this is 0 and not -1. Lets assume I always count the value even in the error case, the results would still be correct.